packages/core/src/storage/diagramObserverStorage.ts

```diff
@@ -0,0 +1,60 @@
+import { ItemValue } from '../types/ItemValue';
+import { LinkId } from '../types/Link';
+import { NodeId } from '../types/Node';
+import { NodeStatus } from '../Executor';
+import { ObserverStorage } from '../types/ObserverStorage';
+
+/**
+ * implementation of ObserverStorage using Maps
+ */
+export class DiagramObserverStorage implements ObserverStorage {
```

Comment:
This class seems very similar to `ExecutionMemory` in that they are maps of items, counts etc. Did we put some thought into how they are different, what separated concerns they have etc?

Reply:
ExecutionMemory: Primarily saves the results of a single run for one diagram, without the need for persistent storage.  
DiagramObserverStorage: Stores the results of multiple runs for multiple diagrams, requiring the data to be saved in other storage spaces.  

Yes, they are somewhat similar at this point. After completing the feature to "update the Vscode Server and SocketServer to accommodate interface changes," I will further refine the implementation of Storage.